src/components/HdLink.vue

```diff
@@ -2,6 +2,8 @@
 export const TYPES = [
   'primary',
   'secondary',
+  'dark',
+  'secondary-dark',
 ];
```

Comment:
Instead of having all these cases, I'd suggest to have just `primary` and `secondary`, and set `color: inherit;` in the `link` mixin. That should handle the dark mode without too much hassle.